org.lflang/src/org/lflang/generator/c/CGenerator.xtend

```diff
@@ -843,6 +844,15 @@ class CGenerator extends GeneratorBase {
             
             // Create docker file.
             if (targetConfig.dockerOptions !== null) {
+                if (isFederated) {
+                    var rtiPath = fileConfig.getSrcGenBasePath().resolve("RTI")
+                    var rtiDir = rtiPath.toFile()
+                    if (!rtiDir.exists()) {
```

Comment:
I didn't realize this before, but with this design, if we change something in the RTI's source code in `c/reactor-c/core/federated/RTI`, it will not show up in the RTI's source code that is in `src-gen/RTI` if this folder already exists, because `copyRtiFiles` is skipped. Perhaps `copyRtiFiles(rtiDir, coreFiles)` should be out of the if statement? Apologies for delaying the merge, but I think this is rather important.

Reply:
that's a good point. I think a better design would be to put the RTI docker file generation outside of the for (federate : federates) loop. I also moved the copyRtiFiles outside the if statement. Thanks for your input!

Reply:
That's a good design! Thanks for addressing this so quickly